scipy/linalg/tests/test_matfuncs.py

```diff
@@ -412,69 +412,64 @@ def test_gh7839(self):
 
     def test_data_size_preservation_uint_in_float_out(self):
         M = np.zeros((10, 10), dtype=np.uint8)
-        # input bit size is 8, but minimum float bit size is 16
-        assert sqrtm(M).dtype == np.float16
+        assert sqrtm(M).dtype == np.float64
         M = np.zeros((10, 10), dtype=np.uint16)
-        assert sqrtm(M).dtype == np.float16
+        assert sqrtm(M).dtype == np.float64
         M = np.zeros((10, 10), dtype=np.uint32)
-        assert sqrtm(M).dtype == np.float32
+        assert sqrtm(M).dtype == np.float64
         M = np.zeros((10, 10), dtype=np.uint64)
         assert sqrtm(M).dtype == np.float64
 
     def test_data_size_preservation_int_in_float_out(self):
         M = np.zeros((10, 10), dtype=np.int8)
-        # input bit size is 8, but minimum float bit size is 16
-        assert sqrtm(M).dtype == np.float16
+        assert sqrtm(M).dtype == np.float64
         M = np.zeros((10, 10), dtype=np.int16)
-        assert sqrtm(M).dtype == np.float16
+        assert sqrtm(M).dtype == np.float64
         M = np.zeros((10, 10), dtype=np.int32)
-        assert sqrtm(M).dtype == np.float32
+        assert sqrtm(M).dtype == np.float64
         M = np.zeros((10, 10), dtype=np.int64)
         assert sqrtm(M).dtype == np.float64
 
     def test_data_size_preservation_int_in_comp_out(self):
         M = np.array([[2, 4], [0, -2]], dtype=np.int8)
-        # input bit size is 8, but minimum complex bit size is 64
-        assert sqrtm(M).dtype == np.complex64
+        assert sqrtm(M).dtype == np.complex128
         M = np.array([[2, 4], [0, -2]], dtype=np.int16)
-        # input bit size is 16, but minimum complex bit size is 64
-        assert sqrtm(M).dtype == np.complex64
+        assert sqrtm(M).dtype == np.complex128
         M = np.array([[2, 4], [0, -2]], dtype=np.int32)
-        assert sqrtm(M).dtype == np.complex64
+        assert sqrtm(M).dtype == np.complex128
         M = np.array([[2, 4], [0, -2]], dtype=np.int64)
         assert sqrtm(M).dtype == np.complex128
 
     def test_data_size_preservation_float_in_float_out(self):
         M = np.zeros((10, 10), dtype=np.float16)
-        assert sqrtm(M).dtype == np.float16
+        assert sqrtm(M).dtype == np.float64
         M = np.zeros((10, 10), dtype=np.float32)
         assert sqrtm(M).dtype == np.float32
         M = np.zeros((10, 10), dtype=np.float64)
         assert sqrtm(M).dtype == np.float64
         if hasattr(np, 'float128'):
             M = np.zeros((10, 10), dtype=np.float128)
-            assert sqrtm(M).dtype == np.float128
+            assert sqrtm(M).dtype == np.float64
 
     def test_data_size_preservation_float_in_comp_out(self):
         M = np.array([[2, 4], [0, -2]], dtype=np.float16)
-        # input bit size is 16, but minimum complex bit size is 64
-        assert sqrtm(M).dtype == np.complex64
+        assert sqrtm(M).dtype == np.complex128
```

Comment:
When the input is `np.float16`, the output is `np.complex128`. Is it OK?

Reply:
All float16 should be upcast to float32.